File: src/test/java/summaryranges/SolutionTest.java
package summaryranges;

import java.util.List;
import java.util.stream.Stream;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

class SolutionTest {

    private final Solution solution = new Solution();

    @ParameterizedTest
    @MethodSource("successTestData")
    public void shouldReturnTrue(int[] nums, List<String> expectedRanges) {
        List<String> ranges = solution.summaryRanges(nums);

        Assertions.assertThat(ranges).containsExactly(expectedRanges.toArray(new String[expectedRanges.size()]));

    }

    private static Stream<Arguments> successTestData() {
        return Stream.of(
            Arguments.of(new int[]{}, List.of()),
            Arguments.of(new int[]{0}, List.of("0")),
            Arguments.of(new int[]{0, 1, 2, 4, 5, 7}, List.of("0->2", "4->5", "7")),
            Arguments.of(new int[]{0, 2, 3, 4, 6, 8, 9}, List.of("0", "2->4", "6", "8->9"))
        );
    }
}

File: src/test/kotlin/dev/sanket/rotatematrix/SolutionTest.kt
package dev.sanket.rotatematrix

import assertk.assertThat
import assertk.assertions.containsExactly
import dev.sanket.dev.sanket.rotatematrix.Solution
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class SolutionTest {
    private val solution = Solution()

    @ParameterizedTest
    @MethodSource("testData")
    fun shouldPass(
        matrix: Array<IntArray>,
        expected: Array<IntArray>,
    ) {
        solution.rotate(matrix)
        matrix.forEachIndexed { index, it ->
            assertThat(it).containsExactly(*expected[index])
        }
    }

    companion object {
        @JvmStatic
        fun testData(): Stream<Arguments> =
            Stream.of(
                Arguments.of(arrayOf(intArrayOf(1)), arrayOf(intArrayOf(1))),
                Arguments.of(arrayOf(intArrayOf(4, 1), intArrayOf(5, 2)), arrayOf(intArrayOf(5, 4), intArrayOf(2, 1))),
                Arguments.of(
                    arrayOf(intArrayOf(1, 2, 3), intArrayOf(4, 5, 6), intArrayOf(7, 8, 9)),
                    arrayOf(intArrayOf(7, 4, 1), intArrayOf(8, 5, 2), intArrayOf(9, 6, 3)),
                ),
            )
    }
}
